File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l = 0, r = 0;
        int sum = 0;
        int n = nums.size();
        int mini = INT_MAX;

        while (r < n) {
            sum += nums[r];  

           
            while (sum >= target) {
                mini = min(mini, r - l + 1);  


                sum -= nums[l]; 
                l++;  
            }
            r++; 
        }
        return (mini != INT_MAX) ? mini : 0;
    }
};

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        minlen = float('inf') 
        currsum = 0
        left = 0

        for right in range(len(nums)):
            currsum += nums[right]

            while currsum >= target:
                minlen = min(minlen, right - left + 1)
                currsum -= nums[left]
                left += 1

        return minlen if minlen != float('inf') else 0
